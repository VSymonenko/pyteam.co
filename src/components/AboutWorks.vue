<template lang="pug">
  v-flex(:style="{ 'flex-direction': $vuetify.breakpoint.xsOnly ? 'column' : reverse ? 'row-reverse' : 'row' }" class='flex-row' style={ maxHeight: '250px'})
    div(class='flex-column' :style={ flex: '1 1 100%', margin: '10px', padding: '0 20px' })
      img(:src='img' v-if='img' :style="{ height: $vuetify.breakpoint.xsOnly ? '50px' : '100%', width: '100%' }")
    div(class='flex-column' :style={ flex: '1 1 100%', justifyContent: 'space-between', margin: '10px', padding: '0 20px' })
      h3(:style={ height: '40px', width: '100%'}) {{ title }}
      p(:style={ margin: 0, width: '100%', display: 'flex' }) {{ description }}
</template>

<script lang="ts">
import Vue from 'vue';
import Component from 'vue-class-component';

@Component({
  name: 'AboutWorks',
  props: {
    title: {
      type: String,
    },
    img: {
      type: String,
    },
    description: {
      type: String,
    },
    reverse: {
      type: Boolean,
    },
  },
})

export default class AboutWorks extends Vue {
}
</script>

<style lang="stylus" scoped>
img
  object-fit cover
</style>
