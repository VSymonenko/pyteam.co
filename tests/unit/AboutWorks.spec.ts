import AboutWorks from '@/components/AboutWorks.vue';
import { shallowMount } from '@vue/test-utils';
import './common';

test('renders correctly', () => {
  const wrapper = shallowMount(AboutWorks);

  expect(wrapper.isVueInstance()).toBeTruthy();
  expect(wrapper).toMatchSnapshot();
});

describe('AboutWorks', () => {
  it('renders title in AboutWorks', () => {
    const title = 'new title';
    const wrapper = shallowMount(AboutWorks, {
      propsData: { title },
    });
    expect(wrapper.props().title).toContain(title);
  });

  it('renders description in AboutWorks', () => {
    const description = 'new description';
    const wrapper = shallowMount(AboutWorks, {
      propsData: { description },
    });
    expect(wrapper.props().description).toContain(description);
  });

  it('renders img src in AboutWorks', () => {
    const img = 'new img';
    const wrapper = shallowMount(AboutWorks, {
      propsData: { img },
    });
    expect(wrapper.props().img).toContain(img);
  });

  it('renders reverse block in AboutWorks', () => {
    const reverse = true;
    const wrapper = shallowMount(AboutWorks, {
      propsData: { reverse },
    });
    expect(wrapper.props().reverse).toBeTruthy();
  });
});
