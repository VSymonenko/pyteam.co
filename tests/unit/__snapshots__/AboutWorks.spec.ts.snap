// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<vcard-stub tag="div" class="flex-row" style="max-height: 250px; flex-direction: row;">
  <div class="flex-column" style="margin: 10px; padding: 0px 20px;">
    <!---->
  </div>
  <div class="flex-column" style="justify-content: space-between; margin: 10px; padding: 0px 20px;">
    <h3 style="height: 40px; width: 100%;"></h3>
    <p style="margin: 0px; width: 100%; display: flex;"></p>
  </div>
</vcard-stub>
`;
